### ISSUE TITLE
[Bug]: Clicking Delete key on keyboard moves the cursor to the end (ComboBox)

### ISSUE THREAD

**Issue by iamvimals** [microsoft/fluentui#31051]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 11 10.0.22621
    CPU: (12) x64 12th Gen Intel(R) Core(TM) i7-1265U
    Memory: 3.38 GB / 15.69 GB
  Browsers:
    Edge: Chromium (123.0.2420.97)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codepen.io/iamvimals/pen/jORpJOK

## Actual Behavior
Steps:
1. Click on the combobox dropdown, and select any option (e.g. option Green considered here).
2. Now move the cursor right before the first 'e' character. You must use the mouse to click before the 'e'. When you use the arrow keys the issue does not repro.
3. Click on the delete key on the keyboard.
4. ISSUE: The first 'e' character gets deleted (which is right), but the cursor moves to the end of the text which is the issue.

## Expected Behavior
The cursor should remain at the same place i.e. where the 'e' character was deleted, and move anywhere.

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

no

### Validations

- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by iamvimals** [comment]
@TristanWatanabe Yes, I can still reproduce this, and you just have to select an option, place the cursor somewhere in the middle, and hit delete key.

https://github.com/microsoft/fluentui/assets/64698374/609ff930-4af9-44ac-b5a1-a0204acc6640

**Comment by sopranopillow** [comment]
Introduced in 8.103.0

**Comment by sijieliMSFT** [comment]
I'm also running into this issue, wondering if there's an estimated time for releasing the fix? Thanks.
